src/components/client/UploadImage/context/UploadImageContext/UploadImageContext.tsx

```diff
@@ -0,0 +1,74 @@
+'use client';
+
+import {
+  ChangeEvent,
+  createContext,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { toast } from 'react-toastify';
+
+import { UploadImageContextProps, UploadImageProviderProps } from './UploadImageContext.type';
+
+export const UploadImageContext = createContext<UploadImageContextProps>({
+  images: [],
+  handleUploadImage: () => {},
+  handleRemoveImage: () => {},
+});
+
+export const UploadImageProvider = ({
+  limit = 1,
+  onChange,
+  children,
+}: UploadImageProviderProps) => {
+  const [images, setImages] = useState<File[]>([]);
+  const onChangeRef = useRef(onChange);
```

Comment:
[ P2 ]
useRef를 꼭 써야하는 이유가없다면
```suggestion
  const onChangeRef = useCallback(onChange,[onChange]);
```
위와같이 해결해주는건어떨까요? 그럼 하단 useEffect도 없어도될것같습니다!
그리고 해당 props에 대해서 메모이제이션은 불필요하지않을까 생각도 조금드네요!

Reply:
음 제가 의도한 거는 onChange 참조 값이 변경되었을 때 의존성에 영향을 최소화할려고 useRef로 감쌌던 것 같습니다. useCallback을 사용해도 참조 값이 변하면 의존성에 영향을 주기 때문에 다른 의도라고 보시면 됩니다!